src/main/java/uk/gov/hmcts/reform/preapi/media/MediaKind.java

```diff
@@ -158,10 +158,19 @@ public PlaybackDTO playAsset(String assetName, String userId) throws Interrupted
     }
 
     private void refreshStreamingLocatorForUser(String userId, String assetName) {
-        mediaKindClient.deleteStreamingLocator(userId);
-
         var now = OffsetDateTime.now();
 
+        // check streaming locator is still valid
+        try {
+            var locator = mediaKindClient.getStreamingLocator(userId);
+            if (locator.getProperties().getEndTime().toInstant().isAfter(now.toInstant())) {
```

Comment:
Probably need to check here that the streaming locator is on the correct asset as well, otherwise you will get an error when you play one asset, then attempt to play a different asset. 

Another workaround for this could be, to change the naming convention of the locators to somethign like `<user-id>_<asset-name>`

Reply:
Looks good to me. However, you will also need to update the playAsset function to use the correct name when getting the streaming locator:
```java
var paths = mediaKindClient.getStreamingLocatorPaths(userId + "_" + assetName); 
```